### ISSUE TITLE
Error in auto_complete for integers

### ISSUE THREAD

**Issue by orrabin** [wvanbergen/scoped_search#118]
When I use `complete_for('some_int > 2')` there is an attempt to find all the values of `some_int` that contain `2` resulting in a sql query with this type of where clause: `WHERE ("my_table"."some_int" LIKE '2)`.
Since `some_int` is an integer field this results in an error.
If I use `complete_for('some_int > 2 ')` (with a space after the '2') it completes to `some_int > 2 or` and `some_int > 2 and` with no error.

This error appears when using complete_for on any integer field.
The use case and logs can be seen here: http://projects.theforeman.org/issues/9088

These are the definitions of search on these integer fields:
```ruby
  scoped_search :in => :discovery_attribute_set, :on => :cpu_count, :rename => :cpu_count, :complete_value => true, :only_explicit => true
  scoped_search :in => :discovery_attribute_set, :on => :disks_size, :rename => :disks_size, :complete_value => true, :only_explicit => true
```

This was tested using scoped_search 2.7.1, activerecord 3.2.21, ruby 1.9.3, rails 3.2.21, PostgreSQL 9.3.5

**Comment by ohadlevy** [comment]
I was able to reproduce in latest scoped search as well(3.1.0)

**Comment by wvanbergen** [comment]
This appears to be working for me in `spec/integration/auto_complete_spec.rb`: `Foo.complete_for('int > 2')`. It returns an empty array.

Would it be possible to construct a failing spec for this?

**Comment by orrabin** [comment]
@wvanbergen change line 38 in `spec/integration/auto_complete_spec.rb` from:
```ruby
scoped_search :on => [:string, :int, :date]
```
to:
``` ruby 
scoped_search :on => [:string, :date]
scoped_search :on => :int, :complete_value => true
```
then when you try `Foo.complete_for('int > 2')` you will get the error: `undefined method `quoted_field' for "\"foos\".\"int\"":String`

**Comment by wvanbergen** [comment]
See #119 for a fix that exposes a new problem.

**Comment by wvanbergen** [comment]
Fixed by #121
